### PR TITLE
ansible: avoid duplicate keys

### PR DESCRIPTION
**Pull request by richardlau**
I'm working on resurrecting https://github.com/nodejs/build/pull/1476 but as a GitHub Actions workflow. 

Current yamllint failures are mostly whitespace and I'll deal with those in a separate PR but this is the one non-whitespace issue found.

cc @AshCripps @sam-github

### CODE REVIEW

File: ansible/roles/jenkins-worker/tasks/main.yml [hunk 1]
```diff
@@ -84,9 +84,8 @@
   when: "'rhel7-s390x' in inventory_hostname"
 
 - name: Firewall | remove firewalld
-  when: has_firewalld.rc == 0
   raw: yum remove -y firewalld
-  when: "'rhel7-s390x' in inventory_hostname"
+  when: "'rhel7-s390x' in inventory_hostname and has_firewalld.rc == 0"
```
Comment (sam-github) [inline ansible/roles/jenkins-worker/tasks/main.yml]: What is the behaviour for duplicate keys? If its "last key wins", then this isn't a refactor, its adding a condition. Did you run it  on a host to confirm it works?

Comment (richardlau) [reply ansible/roles/jenkins-worker/tasks/main.yml]: I don't know what the behaviour is for duplicate keys but the linting is flagging it. I've not run it. Is yum happy to remove something that isn't installed (`has_firewalld.rc` is set in the `Firewall | check for firewalld` check immediately above)?

Comment (sam-github) [reply ansible/roles/jenkins-worker/tasks/main.yml]:
```
% ssh test-ibm-rhel7-s390x-1 sudo yum remove -y firewalld && echo YES
Loaded plugins: product-id, search-disabled-repos, subscription-manager
This system is not registered with an entitlement server. You can use subscription-manager to register.
No Match for argument: firewalld
No Packages marked for removal
YES
```

Comment (richardlau) [reply ansible/roles/jenkins-worker/tasks/main.yml]: So do we need the `check for firewalld` at all then?

Comment (sam-github) [reply ansible/roles/jenkins-worker/tasks/main.yml]: It looks suspiciously unnecessary to me. If its last-key wins, the stat resiult was probably always being ignored. Hard to know without trying, though.

Comment (AshCripps) [reply ansible/roles/jenkins-worker/tasks/main.yml]: Not 100% sure but it seems that last key _does_ win https://github.com/ansible/ansible/pull/56933/files#diff-c495d684aecaea8f523caa00ab4ed915R74-R75

**Comment by AshCripps** [comment]
@richardlau Have you managed to test this yet?

File: ansible/roles/jenkins-worker/tasks/main.yml (hunk 1 above)
Comment (sam-github) [inline ansible/roles/jenkins-worker/tasks/main.yml:88]:
```suggestion
  when: "'rhel7-s390x' in inventory_hostname"
```

**Comment by richardlau** [comment]
Updated to remove the `has_firewalld.rc` check and the task that set it. Have run the playbook against `test-ibm-rhel7-s390x-4`.

**Pull request merged**